ecs-cli/modules/cli/local/ps_app.go

```diff
@@ -61,37 +61,73 @@ const (
 	jsonIndent = "  "
 )
 
-// Ps lists the status of the ECS task containers running locally.
+// Ps lists the status of the ECS task containers running locally as a table.
 //
-// Defaults to listing the container metadata in a table format to stdout. If the --json flag is provided,
-// then output the content as JSON instead.
+// Defaults to listing containers from the local Compose file.
+// If the --all flag is provided, then list all local ECS task containers.
+// If the --json flag is provided, then output the format as JSON instead.
 func Ps(c *cli.Context) {
-	client := docker.NewClient()
+	containers := listContainers(c)
+	displayContainers(c, containers)
+}
 
-	containers := listECSLocalContainers(client)
-	if c.Bool(flags.JsonFlag) {
-		displayAsJSON(containers)
-	} else {
-		displayAsTable(containers)
+func listContainers(c *cli.Context) []types.Container {
+	if !c.Bool(flags.AllFlag) {
+		return listLocalComposeContainers()
 	}
+	// Task containers running locally all have a local label
+	return listContainersWithFilters(filters.NewArgs(
+		filters.Arg("label", ecsLocalLabelKey),
+	))
 }
 
-func listECSLocalContainers(client *client.Client) []types.Container {
+func listLocalComposeContainers() []types.Container {
+	wd, _ := os.Getwd()
+	if _, err := os.Stat(filepath.Join(wd, ecsLocalDockerComposeFileName)); os.IsNotExist(err) {
+		logrus.Fatalf("Compose file %s does not exist in current directory", ecsLocalDockerComposeFileName)
```

Comment:
nit: I prefer to have each function return errors and then have the highest level function log them out with Fatal.

Reply:
I'm the opposite :) 
Whenever possible, I try to define errors out of existence. The idea is taken from [reduce the number of exceptions that must be handled](https://web.stanford.edu/~ouster/cgi-bin/cs190-spring15/lecture.php?topic=errorHandling). If you can remove an exception, then it simplifies all other callers by not having them handle another error.

Reply:
⚔️

Reply:
Also... how do you always have a link handy for every single aspect of code design? 😏

Reply:
I have lots of bookmarks :)